### PR TITLE
Add new flag for k8s api server proxy tls rootca in off-cluster mode

### PR DESCRIPTION
**Pull request by wjiangjay**
fix https://github.com/openshift/console/pull/214#issuecomment-403816788

### CODE REVIEW

**Comment by coreos-ui** [comment]
Can one of the admins verify this patch?

**Comment by coreos-ui** [comment]
Can one of the admins verify this patch?

**Comment by wjiangjay** [comment]
@spadgett help take a look, thanks

File: cmd/bridge/main.go [hunk 1]
```diff
@@ -266,10 +267,28 @@ func main() {
...
+			k8sOffClusterCA := ""
+			if *fK8sModeOffClusterCAFile != "" {
+				k8sOffClusterCA = *fK8sModeOffClusterCAFile
+			} else {
+				k8sOffClusterCA = *fCAFile
+			}
+			rootCAs := x509.NewCertPool()
+			ca, err := ioutil.ReadFile(k8sOffClusterCA)
+			if err != nil {
+				log.Fatalf("Read k8sOffClusterCA fail")
```
Comment (jhadvig) [inline cmd/bridge/main.go]: It's not a good practice to use variable names in `log`, also the `err` needs to be attached to the output
```
log.Fatalf("Failed to read k8s-mode-off-cluster CA file: %v", err)
```

Comment (wjiangjay) [reply cmd/bridge/main.go]: Thanks for your suggestion

File: cmd/bridge/main.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ func main() {
 	fK8sMode := fs.String("k8s-mode", "in-cluster", "in-cluster | off-cluster")
 	fK8sModeOffClusterEndpoint := fs.String("k8s-mode-off-cluster-endpoint", "", "URL of the Kubernetes API server.")
 	fK8sModeOffClusterSkipVerifyTLS := fs.Bool("k8s-mode-off-cluster-skip-verify-tls", false, "DEV ONLY. When true, skip verification of certs presented by k8s API server.")
+	fK8sModeOffClusterCAFile := fs.String("k8s-mode-off-cluster-ca-file", "", "PEM file for Kubernetes API server, will be igrored if k8s-mode-off-cluster-skip-verify-tls is true, and use ca-file if not given")
```
Comment (benjaminapetersen) [inline cmd/bridge/main.go]: This wording could be smoothed out just a bit more.  `and use ca-file if not given` is a bit terse & could be confusing.

Comment (wjiangjay) [reply cmd/bridge/main.go]: Yeah, seems it is confusing

**Comment by wjiangjay** [comment]
@spadgett  help take a look again, thanks

**Comment by spadgett** [comment]
Hi, I'm sorry for leaving this for so long.

I think since this is only used for development and never in production the current behavior is reasonable. We're trying to get away from adding new flags when possible since we're transitioning to config files. If we ever support off-cluster mode for production use, we can revisit this.

/close

**Pull request closed without merging**